File: cs/SolidEdge/ExtractStructure/ExtractStructure/OccurrenceItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExtractStructure
{
    public class OccurrenceItem
    {
        private List<OccurrenceItem> _children = new List<OccurrenceItem>();

        public OccurrenceItem()
        {
        }

        public OccurrenceItem(SolidEdgeAssembly.Occurrence occurrence)
        {
            FileName = occurrence.OccurrenceFileName;
        }

        public OccurrenceItem(SolidEdgeAssembly.SubOccurrence subOccurrence)
        {
            FileName = subOccurrence.SubOccurrenceFileName;
        }

        public string FileName;
        public List<OccurrenceItem> Occurrence { get { return _children; } }

        public override string ToString()
        {
            return FileName;
        }
    }
}

File: vb/SolidEdge/ExtractStructure/ExtractStructure/OccurrenceItem.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace ExtractStructure
	Public Class OccurrenceItem
		Private _children As New List(Of OccurrenceItem)()

		Public Sub New()
		End Sub

		Public Sub New(ByVal occurrence As SolidEdgeAssembly.Occurrence)
			FileName = occurrence.OccurrenceFileName
		End Sub

		Public Sub New(ByVal subOccurrence As SolidEdgeAssembly.SubOccurrence)
			FileName = subOccurrence.SubOccurrenceFileName
		End Sub

		Public FileName As String
		Public ReadOnly Property Occurrence() As List(Of OccurrenceItem)
			Get
				Return _children
			End Get
		End Property

		Public Overrides Function ToString() As String
			Return FileName
		End Function
	End Class
End Namespace
